test/cases/default_value_references_comptime_var.zig

```diff
@@ -0,0 +1,50 @@
+export fn foo() void {
+    comptime var a: u8 = 0;
+    _ = struct { comptime *u8 = &a };
+}
+export fn bar() void {
+    comptime var a: u8 = 0;
+    _ = @Type(.{ .@"struct" = .{
+        .layout = .auto,
+        .fields = &.{.{
+            .name = "0",
+            .type = *u8,
+            .default_value_ptr = @ptrCast(&&a),
+            .is_comptime = true,
+            .alignment = @alignOf(*u8),
+        }},
+        .decls = &.{},
+        .is_tuple = true,
+    } });
+}
+
+export fn baz() void {
+    comptime var a: u8 = 0;
+    _ = struct { foo: *u8 = &a };
+}
+export fn qux() void {
+    comptime var a: u8 = 0;
+    _ = @Type(.{ .@"struct" = .{
+        .layout = .auto,
+        .fields = &.{.{
+            .name = "foo",
+            .type = *u8,
+            .default_value_ptr = @ptrCast(&&a),
+            .is_comptime = false,
+            .alignment = @alignOf(*u8),
+        }},
+        .decls = &.{},
+        .is_tuple = false,
+    } });
+}
+
+// error
+//
+// :1:8: error: field default value contains reference to comptime var
```

Comment:
This source location is inexcusably bad, would you mind trying to fix it :P (feel free to ping me on Zulip if you need a hand)